faraday/data.py

```diff
@@ -0,0 +1,246 @@
+# Imports - General
+import os
+import ConfigParser
+import json
+import base64
+import sys
+import struct
+import logging.config
+import argparse
+import shutil
+
+from flask import Flask
+from flask import request
+
+DATA_FIXED_LEN = 42
+
+# Add Faraday library to the Python path.
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+
+# Imports - Faraday Specific
+from faraday.proxyio import faradaybasicproxyio
+
+
+# Start logging after importing modules
+relpath1 = os.path.join('etc', 'faraday')
+relpath2 = os.path.join('..', 'etc', 'faraday')
+setuppath = os.path.join(sys.prefix, 'etc', 'faraday')
+userpath = os.path.join(os.path.expanduser('~'), '.faraday')
+path = ''
+
+
+for location in os.curdir, relpath1, relpath2, setuppath, userpath:
+    try:
+        logging.config.fileConfig(os.path.join(location, "loggingConfig.ini"))
+        path = location
+        break
+    except ConfigParser.NoSectionError:
+        pass
+
+logger = logging.getLogger('Data')
+
+# Load Telemetry Configuration from telemetry.ini file
+
+#Create Proxy configuration file path
+dataConfigPath = os.path.join(path, "data.ini")
+logger.debug('data.ini PATH: ' + dataConfigPath)
+
+dataConfig = ConfigParser.RawConfigParser()
+dataConfig.read(dataConfigPath)
+
+# Command line input
+parser = argparse.ArgumentParser(description='Provides a generic data server to proxy.')
+parser.add_argument('--init-config', dest='init', action='store_true', help='Initialize Data configuration file')
+parser.add_argument('--flaskhost', help='Set Flask server hostname/address')
+parser.add_argument('--flaskport', help='Set Flask server port')
+parser.add_argument('--proxyhost', help='Set Proxy server hostname/address')
+parser.add_argument('--proxyport', help='Set Proxy server port')
+parser.add_argument('--start', action='store_true', help='Start Data server')
+
+# Parse the arguments
+args = parser.parse_args()
+
+
+def initializeDataConfig():
+    '''
+    Initialize Data configuration file from data.sample.ini
+
+    :return: None, exits program
+    '''
+
+    logger.info("Initializing Data")
+    shutil.copy(os.path.join(path, "data.sample.ini"), os.path.join(path, "data.ini"))
+    logger.info("Initialization complete")
+    sys.exit(0)
+
+
+def configureData(args, simpleuiConfigPath):
```

Comment:
Update `simpleuiConfigPath` parameter and variable when used in this function. Also make sure to update the docstring.